lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
I meant this for a case where we receive consecutive messages without '\0' in them, so when first such message arrives we copy its contents to the buffer in this class that has enough space (with space to spare) or if it doesn't then we replace it with a bigger one and copy, hence on a second and so on half-messages we won't have to copy what we've received before and only append the received data to the buffer we have in class.